src/client.ts

```diff
@@ -94,6 +85,27 @@ export class Client {
     this._assignedRoute = null;
   }
 
+  /**
+   * Registers custom elements used by the client application
+   */
+  public registerCustomElements(): void {
+    const clientInstance = this;
+    class IfcClientLinkElementComplete extends IfcClientLinkElement {
+      constructor() {
+        super(clientInstance);
+      }
+    }
+    class IfcHostLinkElementComplete extends IfcHostLinkElement {
+      constructor() {
+        super(clientInstance);
+      }
+    }
+    clientInstance.addListener('environmentalData', envData => {
```

Comment:
@MattCheely I needed to add a listener to wait for the environmentalData before rendering the link components. I found that in the sample app, the link components would render before the environmentalData loads. Adding a listener here solves that problem, but I would like to know if this is a good solution or not.

Reply:
This is an interesting approach that I hadn't considered. by delaying the definition of the elements until the envData is present, the custom elements on the page won't do anything until the envData is received. Likely that means that the text that would be in the link just shows up as plain unstlyed text until envData arrives.

That makes the links not clickable, but also could create a somewhat jarring styling flash when they convert to links after the envData loads.  I'll outline/suggest another approach on the custom element itself that's closer to my original idea for comparison.  I'm not sure right now which I think is the best approach, I'd need to think about it a bit.